bootstrap.sh

```diff
@@ -0,0 +1,106 @@
+#!/bin/bash
+#
+# This is an easy-bake script to download and build all UnifyCR's dependencies.
+#
+
+ROOT="$(pwd)"
+
+mkdir -p deps
+mkdir -p install
+INSTALL_DIR=$ROOT/install
+
+cd deps
+
+repos=(	git://git.mcs.anl.gov/bmi
+	https://github.com/google/leveldb.git
+	https://github.com/LLNL/GOTCHA.git
+	https://github.com/pmodels/argobots.git
+	https://github.com/mercury-hpc/mercury.git
+	https://xgitlab.cels.anl.gov/sds/margo.git
+	https://github.com/dvidelabs/flatcc.git
+)
+
+for i in "${repos[@]}" ; do
+	# Get just the name of the project (like "mercury")
+	name=$(basename $i | sed 's/\.git//g')
+	if [ -d $name ] ; then
+		echo "$name already exists, skipping it"
+	else
+		if [ "$name" == "mercury" ] ; then
+			git clone --recurse-submodules $i
+		else
+			git clone $i
+		fi
+	fi
+done
+
+echo "### building bmi ###"
+cd bmi
+./prepare && ./configure --enable-shared --enable-bmi-only \
+	--prefix="$INSTALL_DIR"
+make -j $(nproc) && make install
```

Comment:
Does $nproc get set in here, or should the user set that before calling the script?

Reply:
It's not a variable; we're invoking the `nproc` command here to return the number of CPUs:
```
$ nproc
12
```

Reply:
Oh, gotcha.  Thanks.